packages/engine/src/lib/services/progress.service.ts

```diff
@@ -1,20 +1,17 @@
+import { makeHttpRequest } from '@openops/common';
 import { hashUtils, logger } from '@openops/server-shared';
 import { UpdateRunProgressRequest } from '@openops/shared';
-import { Mutex } from 'async-mutex';
+import { AxiosHeaders } from 'axios';
 import { EngineConstants } from '../handler/context/engine-constants';
 import { FlowExecutorContext } from '../handler/context/flow-execution-context';
 import { throwIfExecutionTimeExceeded } from '../timeout-validator';
 
 let lastRequestHash: string | undefined = undefined;
 
-const lock = new Mutex();
-
 export const progressService = {
   sendUpdate: async (params: UpdateStepProgressParams): Promise<void> => {
     throwIfExecutionTimeExceeded();
-    return lock.runExclusive(async () => {
-      await sendUpdateRunRequest(params);
-    });
```

Comment:
fetch() can hang with no timeout so and next requests to update the progress would be blocked by the lock.